mne/channels/channels.py

```diff
@@ -894,35 +897,47 @@ def interpolate_bads(
             "eeg": ("spline", "MNE", "nan"),
             "meg": ("MNE", "nan"),
             "fnirs": ("nearest", "nan"),
+            "ecog": ("spline", "nan"),
+            "seeg": ("spline", "nan"),
         }
         for key in method:
-            _check_option("method[key]", key, ("meg", "eeg", "fnirs"))
+            _check_option("method[key]", key, tuple(valids.keys()))
             _check_option(f"method['{key}']", method[key], valids[key])
         logger.info("Setting channel interpolation method to %s.", method)
         idx = _picks_to_idx(self.info, list(method), exclude=(), allow_empty=True)
         if idx.size == 0 or len(pick_info(self.info, idx)["bads"]) == 0:
             warn("No bad channels to interpolate. Doing nothing...")
             return self
+        for ch_type in method.copy():
+            idx = _picks_to_idx(self.info, ch_type, exclude=(), allow_empty=True)
+            if len(pick_info(self.info, idx)["bads"]) == 0:
+                method.pop(ch_type)
         logger.info("Interpolating bad channels.")
-        origin = _check_origin(origin, self.info)
+        if not all(np.array(list(method.values())) == "nan") and set(
+            method.keys()
+        ) - set(["seeg"]):
+            origin = _check_origin(origin, self.info)
+        for ch_type, interp in method.items():
+            if interp == "nan":
+                _interpolate_bads_nan(self, ch_type, exclude=exclude)
         if method.get("eeg", "") == "spline":
             _interpolate_bads_eeg(self, origin=origin, exclude=exclude)
-            eeg_mne = False
-        elif "eeg" not in method:
-            eeg_mne = False
-        else:
-            eeg_mne = True
-        if "meg" in method or eeg_mne:
+        if method.get("meg", "") == "MNE" or method.get("eeg", "") == "MNE":
             _interpolate_bads_meeg(
                 self,
                 mode=mode,
+                meg=method.get("meg", "") == "MNE",
+                eeg=method.get("eeg", "") == "MNE",
```

Comment:
[nitpick] `method.get("eeg", "")` is called 3 times and `method.get("meg", "") == "MNE"` is called twice, maybe worth intermediate variables like

```py
eeg_meth = method.get("eeg", "")
meg_use_mne = method.get("meg", "") == "MNE"
```